src/main/java/org/folio/circulation/domain/policy/CirculationPolicyRepository.java

```diff
@@ -70,8 +76,8 @@
       .thenApply(result -> result.next(this::mapToPolicy));
   }
 
-  public CompletableFuture<Result<String>> lookupPolicyId(Item item, User user) {
-    CompletableFuture<Result<String>> findLoanPolicyCompleted = new CompletableFuture<>();
+  public CompletableFuture<Result<Pair<String, List<String>>>> lookupPolicyId(Item item, User user) {
```

Comment:
I think a specific type would better express the return type of this method.

What does the list of strings represent?

Reply:
The list of strings represents conditions of current circulation rule, e.g. "ItemType", "LoanType", "PatronGroup"

Reply:
I think this would be better represented by fields / properties for `item type` and `loan type` (and `patron group` though I don't know why that is needed).